packages/vm/src/context.rs

```diff
@@ -288,6 +327,62 @@ mod test {
         assert!(get_context_data::<S, Q>(ctx).iterators.contains_key(&id3));
     }
 
+    #[test]
+    fn with_func_from_context_works() {
+        let mut instance = make_instance();
+        leave_default_data(instance.context_mut());
+
+        let instance_ptr = NonNull::from(&mut instance);
+        set_wasmer_instance::<S, Q>(instance.context_mut(), instance_ptr);
```

Comment:
I was thinking about that too, but how to test `with_func_from_context_fails_for_missing_instance`? Then `set_wasmer_instance` needs to allow unsetting the value. Better ideas?

Reply:
I mean, it would be acceptable to have a function that clears it, i think... you could call it from the `destroy_unmanaged_context_data` method or something, to disable complaints about it not being used. I don't have a much better idea here...

Reply:
Okay, I just open up `set_wasmer_instance` to take the full type `Option<NonNull<wasmer_runtime_core::instance::Instance>>`. Then we get clearing for free.